src/iexamine.cpp

```diff
@@ -2280,6 +2280,179 @@ void iexamine::tree_hickory(player &p, const tripoint &examp)
     none( p, examp );
 }
 
+void iexamine::tree_maple(player &p, const tripoint &examp)
+{
+    if( !p.has_items_with_quality( "DRILL", 1, 1 ) ) {
+        add_msg( m_info, _( "You need a tool to drill the crust to tap this maple tree." ) );
+        return;
+    }
+
+    if( !p.has_items_with_quality( "HAMMER", 1, 1 ) ) {
+        add_msg( m_info, _( "You need a tool to hammer the spile into the crust to tap this maple tree." ) );
+        return;
+    }
+
+    std::string spile_name = item( "maple_tree_spile", 0 ).tname( 1 );
+    std::string title = string_format( _( "Which %s:" ), spile_name.c_str() );
+
+    auto spile_loc = g->inv_map_splice( []( const item &it ) { return it.typeId() == "maple_tree_spile"; }, title, 1 );
```

Comment:
If they don't differ - and it doesn't look like they do because you're checking them by type id - it would be better to just pick the first one and not bother the player with selecting it. You can do it with `p.has_amount(id_here, 1)` and `p.use_amount(id_here, 1)`.